stable/redis-ha/templates/redis-haproxy-servicemonitor.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) ( .Values.haproxy.metrics.serviceMonitor.enabled ) ( .Values.haproxy.metrics.enabled ) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+{{- with .Values.haproxy.metrics.serviceMonitor.labels }}
+  labels: {{ toYaml . | nindent 4}}
```

Comment:
This won't work, once you add a label you're gonna end up with this:
```YAML
labels: foo: bar
  bar: foo
```

I would actually change it to:
```YAML
labels:
{{- with .Values.haproxy.metrics.serviceMonitor.labels }}
{{ toYaml . | nindent 4}}
{{- end }}
```

If `labels` is empty the yaml is still valid and we make sure that our labels have the correct indentation.

Reply:
No, `nindent` would add a newline at fisrt, we'll got what you've desribed with `indent`, I've tried, but maybe I should remove whitespace after `:`.
```
helm template ./ -x templates/redis-haproxy-servicemonitor.yaml       15:28:34  
---
# Source: redis-ha/templates/redis-haproxy-servicemonitor.yaml

apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels: 
    test: label
    
  name: release-name-redis-ha-haproxy
spec:
  endpoints:
  - targetPort: 9101
  jobLabel: release-name-redis-ha-haproxy
  namespaceSelector:
    matchNames:
    - default
  selector:
    matchLabels:
      app: redis-ha
      release: release-name
      component: redis-ha-haproxy
```

Reply:
Oh you're right! 

Awesome then, I will let one of the owner of the chart approve this PR.

Thanks for you changes!

Reply:
Or even move to the separate line with simple `indent` to not confuse anyone

Reply:
I'm fine with both I don't really mind